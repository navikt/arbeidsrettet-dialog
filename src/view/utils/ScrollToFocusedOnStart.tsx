import React, { ReactElement, useEffect } from 'react';
import { RouteComponentProps, RouteProps, withRouter } from 'react-router-dom';

const ScrollToFocusedOnStart = (props: RouteProps): ReactElement<RouteComponentProps<any>> => {
    useEffect(() => document.activeElement?.scrollIntoView(false), []);
    return <>{props.children}</>;
};

export default withRouter(ScrollToFocusedOnStart);
