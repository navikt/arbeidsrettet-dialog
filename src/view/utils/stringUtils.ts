export const cutStringAtLength = (text: string, maxLength: number, postfix?: string) => {
    return text.length > maxLength
        ? `${text.substring(0, postfix ? maxLength - postfix.length : maxLength)}${postfix ? postfix : ''}`
        : text;
};
