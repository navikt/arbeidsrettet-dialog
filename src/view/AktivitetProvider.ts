import React from 'react';

import { Status, hasError } from '../api/typer';
import { AktivitetApi } from '../api/UseApiBasePath';
import { Aktivitet, ArenaAktivitet } from '../utils/aktivitetTypes';
import { fetchData } from '../utils/Fetch';
import { StringOrNull } from '../utils/Typer';
import { hentAktiviteterGraphql } from '../api/aktivitetsplanGraphql';
import { createGenericStore } from '../utils/genericStore';
import { useShallow } from 'zustand/react/shallow';

export type MaybeAktivitet = Aktivitet | ArenaAktivitet | undefined;

export interface AktivitetDataProviderType {
    aktiviteterStatus: Status;
    arenaAktiviteterStatus: Status;
    aktiviteter: Aktivitet[];
    arenaAktiviteter: ArenaAktivitet[];
}

export interface AktivitetState {
    aktiviteterStatus: Status;
    arenaAktiviteterStatus: Status;
    aktiviteter: Aktivitet[];
    arenaAktiviteter: ArenaAktivitet[];
}

export const AktivitetContext = React.createContext<AktivitetDataProviderType>({
    aktiviteterStatus: Status.INITIAL,
    arenaAktiviteterStatus: Status.INITIAL,
    aktiviteter: new Array<Aktivitet>(),
    arenaAktiviteter: new Array<ArenaAktivitet>()
});

export const useAktivitetContext = () => useAktivitetDataProvider();
export const useHentAktiviteter = () => {
    const hentVeilarbAktiviteter = useAktivitetStore(useShallow((state) => state.fetch));
    const hentTiltaksAktiviteter = useTiltaksAktivitetStore(useShallow((state) => state.fetch));
    return (fnr: string | undefined) => {
        hentVeilarbAktiviteter(fnr);
        hentTiltaksAktiviteter(fnr);
    };
};

export function harAktivitetDataFeil(aktivitetData: AktivitetDataProviderType, arenaAktivitet: boolean): boolean {
    if (arenaAktivitet) {
        return hasError(aktivitetData.arenaAktiviteterStatus);
    } else {
        return hasError(aktivitetData.aktiviteterStatus);
    }
}

export const findAktivitet = (aktivitetData: AktivitetDataProviderType, aktivitetId?: StringOrNull): MaybeAktivitet => {
    if (!aktivitetId) return undefined;

    const aktiviteterHasData =
        aktivitetData.aktiviteterStatus === Status.OK || aktivitetData.aktiviteterStatus === Status.RELOADING;
    const aktiviteter: Aktivitet[] = aktiviteterHasData ? aktivitetData.aktiviteter : [];

    const arenaHasData =
        aktivitetData.arenaAktiviteterStatus === Status.OK || aktivitetData.arenaAktiviteterStatus === Status.RELOADING;
    const arena: ArenaAktivitet[] = arenaHasData ? aktivitetData.arenaAktiviteter : [];

    return [...aktiviteter, ...arena].find((aktivitet) => aktivitet.id === aktivitetId);
};

const arenaAktivitetUrl = AktivitetApi.hentArenaAktiviteter;

const hentVeilarbAktiviteter = (fnr: string | undefined) =>
    hentAktiviteterGraphql(fnr).then((response) => response.data.perioder.flatMap((periode) => periode.aktiviteter));
export const useAktivitetStore = createGenericStore<undefined | Aktivitet[], string | undefined, Aktivitet[]>(
    undefined,
    hentVeilarbAktiviteter
);
const hentTiltaksAktiviteter = (fnr: string | undefined) =>
    fetchData<ArenaAktivitet[]>(arenaAktivitetUrl, {
        method: 'POST',
        body: JSON.stringify({ fnr })
    });
export const useTiltaksAktivitetStore = createGenericStore<
    undefined | ArenaAktivitet[],
    string | undefined,
    ArenaAktivitet[]
>(undefined, hentTiltaksAktiviteter);

export const useAktivitetDataProvider = (): AktivitetDataProviderType => {
    const { aktiviteter, aktiviteterStatus } = useAktivitetStore(
        useShallow((state) => ({ aktiviteterStatus: state.status, aktiviteter: state.data }))
    );
    const { arenaAktiviteter, arenaAktiviteterStatus } = useTiltaksAktivitetStore(
        useShallow((state) => ({ arenaAktiviteterStatus: state.status, arenaAktiviteter: state.data }))
    );
    return {
        aktiviteterStatus,
        arenaAktiviteterStatus,
        aktiviteter: aktiviteter || [],
        arenaAktiviteter: arenaAktiviteter || []
    };
};
