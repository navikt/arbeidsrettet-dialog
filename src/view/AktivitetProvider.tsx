import useFetch, { FetchResult, Status, hasData } from '@nutgaard/use-fetch';
import React, { useContext } from 'react';

import { Aktivitet, ArenaAktivitet } from '../utils/AktivitetTypes';
import { REQUEST_CONFIG, fnrQuery } from '../utils/Fetch';
import { StringOrNull } from '../utils/Typer';

export type MaybeAktivitet = Aktivitet | ArenaAktivitet | undefined;

interface AktivitetResponse {
    aktiviteter: Aktivitet[];
}
export interface AktivitetContextType {
    aktiviteter: FetchResult<AktivitetResponse>;
    arenaAktiviter: FetchResult<ArenaAktivitet[]>;
}

const inital: AktivitetContextType = {
    aktiviteter: {
        status: Status.INIT,
        statusCode: 0,
        rerun(): void {}
    },
    arenaAktiviter: {
        status: Status.INIT,
        statusCode: 0,
        rerun(): void {}
    }
};

export const AktivitetContext = React.createContext(inital);
export const useAktivitetContext = () => useContext(AktivitetContext);

export function isLoadingData(aktivitetData: AktivitetContextType): boolean {
    const aktiviteter = hasData(aktivitetData.aktiviteter);
    const arena = hasData(aktivitetData.arenaAktiviter);

    return !aktiviteter || !arena;
}

export function findAktivitet(aktivitetData: AktivitetContextType, aktivitetId?: StringOrNull): MaybeAktivitet {
    if (!aktivitetId) {
        return undefined;
    }

    const aktiviteter = hasData(aktivitetData.aktiviteter) ? aktivitetData.aktiviteter.data.aktiviteter : undefined;
    const arena = hasData(aktivitetData.arenaAktiviter) ? aktivitetData.arenaAktiviter.data : undefined;

    if (aktivitetId.startsWith('ARENA')) {
        return arena && arena.find((a) => a.id === aktivitetId);
    }
    return aktiviteter && aktiviteter.find((a) => a.id === aktivitetId);
}

interface Props {
    fnr?: string;
    apiBasePath: string;
    children: React.ReactNode;
}

export function AktivitetProvider(props: Props) {
    const query = fnrQuery(props.fnr);
    const apiBasePath = props.apiBasePath;

    const aktiviteterFetch = useFetch<AktivitetResponse>(
        `${apiBasePath}/veilarbaktivitet/api/aktivitet${query}`,
        REQUEST_CONFIG
    );
    const arenaAktiviteterFetch = useFetch<ArenaAktivitet[]>(
        `${apiBasePath}/veilarbaktivitet/api/aktivitet/arena${query}`,
        REQUEST_CONFIG
    );

    return (
        <AktivitetContext.Provider value={{ aktiviteter: aktiviteterFetch, arenaAktiviter: arenaAktiviteterFetch }}>
            {props.children}
        </AktivitetContext.Provider>
    );
}
