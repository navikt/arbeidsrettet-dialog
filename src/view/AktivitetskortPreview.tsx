import React from 'react';
import { Aktivitet, DialogData } from '../utils/typer';
import { Undertekst, Undertittel } from 'nav-frontend-typografi';

import useFetch from '../utils/use-fetch';
import { LenkepanelBase } from 'nav-frontend-lenkepanel';
import { ReactComponent as AktivitetsIkon } from './aktivitet_lest.svg';
import { formaterDate, konverterMinutterTilTimer } from '../utils/date';
import Lenke from 'nav-frontend-lenker';

interface Props {
    dialog: DialogData;
}

export function AktivitetskortPreview(props: Props) {
    const aktiviteter = useFetch<Aktivitet[]>('/veilarbaktivitet/api/aktivitet');
    if (aktiviteter.data !== null) {
        const aktivitet = aktiviteter.data.find(aktivitet => aktivitet.id === props.dialog.aktivitetId);

        if (aktivitet) {
            const info = mapAktivitetTypeToPreview(aktivitet);
            return (
                <LenkepanelBase href={'fisk'} className="aktivitetskortpreview">
                    <div className="aktivitetskortpreview__internal-div">
                        <div className="textdiv">
                            <Undertittel children={aktivitet.tittel} className="aktivitetskortpreview__info" />
                            <Undertekst children={info} className="aktivitetskortpreview__info" />
                        </div>
                        <Lenke href={'aktivitetsplan'} className="aktivitetskortpreview__les-mer" children="Les mer" />
                    </div>
                </LenkepanelBase>
            );
        }
    }
    return null;
}

function mapAktivitetTypeToPreview(aktivitet: Aktivitet): string {
    switch (aktivitet.type) {
        case 'STILLING':
            return `Stilling | ${aktivitet.arbeidsgiver}`;
        case 'MOTE':
            return `Møte med NAV | ${formaterDate(aktivitet.fraDato)} | ${konverterMinutterTilTimer(
                aktivitet.klokkeslett!
            )}`;
        case 'SOKEAVTALE':
            return `${formaterDate(aktivitet.fraDato)} - ${formaterDate(aktivitet.tilDato)}`;
        case 'BEHANDLING':
            return aktivitet.behandlingType!;
        case 'SAMTALEREFERAT':
            return `Samtalereferat | ${formaterDate(aktivitet.fraDato)}`;
        case 'EGEN':
            return `Jobbrettet egenaktivitet`;
        case 'IJOBB':
            return `Jobb jeg har nå | ${aktivitet.arbeidsgiver}`;
        case 'TILTAKSAKTIVITET':
            return 'Tiltak gjennom NAV';
        case 'GRUPPEAKTIVITET':
            return 'Gruppeaktivitet';
        case 'UTDANNING':
            return 'Utdanning';
    }

    return '';
}
