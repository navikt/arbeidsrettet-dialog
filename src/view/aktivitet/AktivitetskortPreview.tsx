import React from 'react';
import { DialogData } from '../../utils/Typer';
import { Aktivitet, AktivitetTypes, ArenaAktivitet, ArenaAktivitetTypes } from '../../utils/AktivitetTypes';
import { Undertekst, Undertittel } from 'nav-frontend-typografi';

import UseFetch from '../../utils/UseFetch';
import { LenkepanelBase } from 'nav-frontend-lenkepanel';
import { formaterDate, getKlokkeslett } from '../../utils/Date';
import { aktivitetLenke } from './Aktivitetskort';
import styles from './AktivitetskortPreview.module.less';
import { getTypeText } from './TextUtils';

interface Props {
    dialog: DialogData;
}

export function AktivitetskortPreview(props: Props) {
    const fetch = UseFetch<Aktivitet[]>('/veilarbaktivitet/api/aktivitet');
    const aktiviteter = fetch.data;

    if (!aktiviteter) return null;

    const aktivitet = aktiviteter.find(aktivitet => aktivitet.id === props.dialog.aktivitetId);
    if (!aktivitet) return null;

    const info = getInfoText(aktivitet);
    return (
        <LenkepanelBase href={aktivitetLenke(aktivitet.id)} className={styles.lenkepanelbase}>
            <div>
                <Undertittel className={styles.tittel}>{aktivitet.tittel}</Undertittel>
                <Undertekst>{info}</Undertekst>
            </div>
            <p className={styles.lesmer} children="Les mer" />
        </LenkepanelBase>
    );
}

export function getInfoText(aktivitet: Aktivitet | ArenaAktivitet): string {
    const typeTekst = getTypeText(aktivitet.type);
    switch (aktivitet.type) {
        case AktivitetTypes.STILLING:
            return `${typeTekst} / ${aktivitet.arbeidsgiver}`;
        case AktivitetTypes.MOTE:
            return `${typeTekst} / ${formaterDate(aktivitet.fraDato)} / ${getKlokkeslett(aktivitet.fraDato)}`;
        case AktivitetTypes.SOKEAVTALE:
            return `${formaterDate(aktivitet.fraDato)} - ${formaterDate(aktivitet.tilDato)}`;
        case AktivitetTypes.BEHANDLING:
            return aktivitet.behandlingType ? aktivitet.behandlingType : '';
        case AktivitetTypes.SAMTALEREFERAT:
            return `${typeTekst} / ${formaterDate(aktivitet.fraDato)}`;
        case AktivitetTypes.IJOBB:
            return `${typeTekst} / ${aktivitet.arbeidsgiver}`;
        case ArenaAktivitetTypes.TILTAKSAKTIVITET ||
            ArenaAktivitetTypes.UTDANNINGSAKTIVITET ||
            ArenaAktivitetTypes.GRUPPEAKTIVITET ||
            AktivitetTypes.EGEN:
            return typeTekst;
    }

    return '';
}
