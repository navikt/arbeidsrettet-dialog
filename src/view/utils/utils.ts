import { Aktivitet, AktivitetTypes, AlleAktivitetTypes, ArenaAktivitet } from '../../utils/aktivitetTypes';
import { pathnamePrefix } from '../../utils/UseApiBasePath';

export const erArenaAktivitet = (aktivitetId: string | null | undefined): boolean =>
    !!aktivitetId && aktivitetId.startsWith('ARENA');

export const getBasename = (fnr?: string): string => {
    return settSammenmedSlasher(pathnamePrefix, fnr);
};

export const settSammenmedSlasher = (...ss: Array<string | undefined>): string => {
    const slasherImellom = ss
        .filter((s: string | undefined): s is string => !!s)
        .flatMap<string>((s) => s.split('/'))
        .filter((s) => s.trim() !== '')
        .join('/');
    if (slasherImellom !== '') return '/' + slasherImellom;
    else return '';
};

const erGCP = (): boolean => window.location.hostname.endsWith('intern.nav.no');

export const getContextPath = (): string => (erGCP() ? '' : '/veilarbpersonflatefs');

export const getAktivitetType = (aktivitet: Aktivitet | ArenaAktivitet): AlleAktivitetTypes => {
    if (aktivitet.type === AktivitetTypes.EKSTERN_AKTIVITET) {
        return aktivitet.eksternAktivitet!!.type;
    }
    return aktivitet.type;
};

export const runningOnGithubPages = process.env.REACT_APP_USE_HASH_ROUTER === 'true';
