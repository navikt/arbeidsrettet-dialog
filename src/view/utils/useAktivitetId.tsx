import queryString from 'query-string';
import { useLocation, useParams } from 'react-router';

import { DialogData } from '../../utils/Typer';
import { MaybeAktivitet, findAktivitet, useAktivitetContext } from '../AktivitetProvider';
import { useDialogContext } from '../DialogProvider';

function getFirst(maybeArray: string | (string | null)[]): string | undefined {
    if (Array.isArray(maybeArray)) {
        const first = maybeArray[0];
        return first ?? undefined;
    }
    return maybeArray;
}

export function useAktivitetId(): string | undefined {
    const location = useLocation();
    const { aktivitetId } = queryString.parse(location.search);
    return aktivitetId ? getFirst(aktivitetId) : undefined;
}

export const useSelectedAktivitet = (): MaybeAktivitet => {
    const dialog = useSelectedDialog();
    if (!dialog) return;
    const aktivitetData = useAktivitetContext();
    const aktivitetId = useAktivitetId() ?? dialog?.aktivitetId;
    return findAktivitet(aktivitetData, aktivitetId);
};
export const useSelectedDialog = (): DialogData | undefined => {
    const { dialoger } = useDialogContext();
    const params = useParams();
    const { dialogId } = params;
    return dialoger.find((dialog) => dialog.id === dialogId);
};
