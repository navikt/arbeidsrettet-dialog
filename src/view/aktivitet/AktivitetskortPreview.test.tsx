import { render } from '@testing-library/react';
import React from 'react';
import { MemoryRouter } from 'react-router';

import { Status } from '../../api/typer';
import { Aktivitet, AktivitetStatus, AktivitetTypes, KanalTypes } from '../../utils/aktivitetTypes';
import * as AktivitetProvider from '../AktivitetProvider';
import { AktivitetDataProviderType } from '../AktivitetProvider';
import { AktivitetskortPreview, getInfoText } from './AktivitetskortPreview';

describe('getInfoText', () => {
    it('skal returnere korrekt tekst for stillingsaktivitet', () => {
        const aktivitet: any = {
            type: AktivitetTypes.STILLING,
            tilDato: '2019-10-24T15:44:21.993+02:00',
            arbeidsgiver: 'Testesen'
        };

        const text = getInfoText(aktivitet);
        expect(text).toEqual('Testesen');
    });

    it('skal returnere korrekt tekst for møteaktivitet', () => {
        const aktivitet: any = {
            type: AktivitetTypes.MOTE,
            fraDato: '2019-10-24T15:44:21.993+02:00'
        };

        const text = getInfoText(aktivitet);
        expect(text).toEqual('24.10.2019 / 15:44');
    });

    it('skal returnere korrekt tekst for søkeavtale', () => {
        const aktivitet: any = {
            type: AktivitetTypes.SOKEAVTALE,
            tilDato: '2019-10-25T15:44:21.993+02:00',
            fraDato: '2019-10-24T15:44:21.993+02:00'
        };

        const text = getInfoText(aktivitet);
        expect(text).toEqual('24.10.2019 - 25.10.2019');
    });

    it('skal returnere korrekt tekst for medisinsk behandling', () => {
        const aktivitet: any = {
            type: AktivitetTypes.BEHANDLING,
            behandlingType: 'Kiropraktor'
        };

        const text = getInfoText(aktivitet);
        expect(text).toEqual('Kiropraktor');
    });

    it('skal returnere korrekt tekst for møteaktivitet', () => {
        const aktivitet: any = {
            type: AktivitetTypes.SAMTALEREFERAT,
            fraDato: '2019-10-24T15:44:21.993+02:00'
        };

        const text = getInfoText(aktivitet);
        expect(text).toEqual('24.10.2019');
    });
});

const aktivitet: Aktivitet = {
    adresse: null,
    ansettelsesforhold: null,
    antallStillingerIUken: null,
    antallStillingerSokes: null,
    arbeidssted: null,
    arbeidstid: null,
    avsluttetKommentar: null,
    avtaleOppfolging: null,
    avtalt: false,
    behandlingOppfolging: null,
    behandlingSted: null,
    behandlingType: null,
    beskrivelse: null,
    effekt: null,
    endretAv: null,
    endretDato: null,
    erReferatPublisert: false,
    etikett: null,
    forberedelser: null,
    fraDato: null,
    hensikt: null,
    historisk: false,
    jobbStatus: null,
    kanal: KanalTypes.INTERNETT,
    kontaktperson: null,
    endretAvType: null,
    lenke: null,
    malid: null,
    oppfolging: null,
    opprettetDato: '',
    referat: null,
    status: AktivitetStatus.AVBRUTT,
    stillingsTittel: null,
    transaksjonsType: null,
    versjon: null,
    tittel: 'Kantinemedarbeider',
    id: '123',
    type: AktivitetTypes.STILLING,
    tilDato: '2019-10-24T15:44:21.993+02:00',
    arbeidsgiver: 'Testesen'
};

const aktivitetRes: AktivitetDataProviderType = {
    aktiviteterStatus: Status.OK,
    arenaAktiviteterStatus: Status.OK,
    aktiviteter: [aktivitet],
    arenaAktiviteter: [],
    hentAktiviteter: () => Promise.resolve([aktivitet]),
    hentArenaAktiviteter: () => Promise.resolve([])
};

describe('<AktivitetskortPreview />', () => {
    it('skal rendre stillingsaktivitet som i snapshot', () => {
        vi.spyOn(AktivitetProvider, 'useAktivitetContext').mockImplementation(() => aktivitetRes);

        const wrapper = render(
            <MemoryRouter>
                <AktivitetskortPreview aktivitetId={'123'} />
            </MemoryRouter>
        );

        expect(wrapper.baseElement).toMatchSnapshot();
    });
});
