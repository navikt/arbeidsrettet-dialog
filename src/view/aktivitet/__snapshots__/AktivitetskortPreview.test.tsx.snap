// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AktivitetskortPreview /> skal rendre stillingsaktivitet som i snapshot 1`] = `
<AktivitetskortPreview
  aktivitetId="123"
>
  <section
    aria-label="Aktivitet knyttet til dialog"
  >
    <Systemtittel
      className="visually-hidden"
    >
      <TypografiBase
        className="visually-hidden"
        type="systemtittel"
      >
        <h2
          className="typo-systemtittel visually-hidden"
        >
          Aktivitet knyttet til dialog
        </h2>
      </TypografiBase>
    </Systemtittel>
    <Undertittel>
      <TypografiBase
        type="undertittel"
      >
        <h2
          className="typo-undertittel"
        >
          Stilling
          : 
          Kantinemedarbeider
        </h2>
      </TypografiBase>
    </Undertittel>
    <LenkepanelBase
      border={false}
      href="/aktivitetsplan/aktivitet/vis/123"
      linkCreator={[Function]}
      onClick={[Function]}
    >
      <a
        className="lenkepanel"
        href="/aktivitetsplan/aktivitet/vis/123"
        onClick={[Function]}
      >
        <div>
          <div>
            <EtikettLiten>
              <TypografiBase
                type="etikettLiten"
              >
                <p
                  className="typo-etikett-liten"
                >
                  Stilling
                </p>
              </TypografiBase>
            </EtikettLiten>
            <Undertittel
              id="dialog_header_2"
            >
              <TypografiBase
                id="dialog_header_2"
                type="undertittel"
              >
                <h2
                  className="typo-undertittel"
                  id="dialog_header_2"
                >
                  Kantinemedarbeider
                </h2>
              </TypografiBase>
            </Undertittel>
            <Undertekst>
              <TypografiBase
                type="undertekst"
              >
                <p
                  className="typo-undertekst"
                >
                  Testesen
                </p>
              </TypografiBase>
            </Undertekst>
          </div>
          <p>
            GÃ¥ til aktiviteten
          </p>
        </div>
        <span
          className="lenkepanel__indikator"
          key="indikator"
        />
      </a>
    </LenkepanelBase>
  </section>
</AktivitetskortPreview>
`;
